pic-sure-auth-services/src/main/java/edu/harvard/hms/dbmi/avillach/auth/service/impl/TokenService.java

```diff
@@ -208,49 +230,36 @@ private TokenInspection validateToken(Map<String, Object> inputMap) throws Illeg
     public Map<String, String> refreshToken(String authorizationHeader) {
         logger.debug("RefreshToken starting...");
 
-        Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
```

Comment:
You're using a Map<String, String> to hack a polymorphic response body instead of creating the appropriate classes and interface? And it's a public method? Really?

Reply:
The map is a hold over from legacy code. I will do the work to provide an appropriate classes.